### PR TITLE
apiextensions: improve patch integration test

### PR DESCRIPTION
**Pull request by sttts**
Depends on https://github.com/kubernetes/kubernetes/pull/67635 for being green.

### CODE REVIEW

File: staging/src/k8s.io/apiextensions-apiserver/test/integration/basic_test.go [hunk 1]
```diff
@@ -711,6 +728,26 @@ func TestPatch(t *testing.T) {
 	}
 }
 
+func expectInt64(t *testing.T, obj map[string]interface{}, value int64, pth ...string) {
```
Comment (nikhita) [inline staging/src/k8s.io/apiextensions-apiserver/test/integration/basic_test.go]: can we add this in [`helpers.go`](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/helpers.go)? Looks super useful honestly.

File: staging/src/k8s.io/apiextensions-apiserver/test/integration/basic_test.go [hunk 2]
```diff
@@ -667,29 +668,45 @@ func TestPatch(t *testing.T) {
...
 		t.Fatal(err)
 	}
 
+	t.Logf("Patching .num.num2 to 999")
 	patch := []byte(`{"num": {"num2":999}}`)
 	createdNoxuInstance, err = noxuNamespacedResourceClient.Patch("foo", types.MergePatchType, patch, metav1.UpdateOptions{})
 	if err != nil {
 		t.Fatalf("unexpected error: %v", err)
 	}
+	expectInt64(t, createdNoxuInstance.UnstructuredContent(), 999, "num", "num2")
```
Comment (nikhita) [inline staging/src/k8s.io/apiextensions-apiserver/test/integration/basic_test.go]: we should also add this for `TestSubresourcePatch`

Comment (sttts) [reply staging/src/k8s.io/apiextensions-apiserver/test/integration/basic_test.go]: Can you do that in a follow-up? Feel free to move the helpers.

Comment (nikhita) [reply staging/src/k8s.io/apiextensions-apiserver/test/integration/basic_test.go]: sure.

**Comment by nikhita** [comment]
> Depends on #67635 for being green.

This will be green once https://github.com/kubernetes/kubernetes/pull/67562 gets merged. https://github.com/kubernetes/kubernetes/pull/67635 is not really required.

/release-note-none
/remove-sig scheduling
/area test
/kind cleanup

**Comment by nikhita** [comment]
Picked into https://github.com/kubernetes/kubernetes/pull/67562. Closing this.

/close

**Pull request closed without merging**